lib/taxonomy/mapping.rb

```diff
@@ -0,0 +1,39 @@
+class Taxonomy::Mapping
+  def initialize
+    @all_taxons = Taxonomy::TopicTaxonomy.new.all_taxons
+  end
+
+  def legacy_mapping_for_taxons(taxon_content_ids)
+    all_legacy_associations = taxon_content_ids
+                                .flat_map do |taxon_content_id|
+      legacy_mapping_for_taxon(taxon_content_id).values.flatten
+    end
+
+    all_legacy_associations.uniq do |legacy_association|
+      legacy_association["content_id"]
+    end
+  end
+
+  def legacy_mapping_for_taxon(start_taxon_content_id)
+    start_taxon = @all_taxons.find do |taxon|
+      taxon.content_id == start_taxon_content_id
+    end
+
+    document_types = %w(policy policy_area topic)
+    result = start_taxon.legacy_mapping.slice(*document_types)
+
+    start_taxon.ancestors.each do |ancestor|
+      document_types.each do |document_type|
+        # Look up through the ancestors to find a mapping for each
+        # document_type, but stop once one is found
```

Comment:
It works, because of the `blank?` check, but maybe the comment is misleading, as it doesn't really stop, it just doesn't ignores everything once a value is found.

I'll have a go at rewriting it to make it clearer.

Reply:
I've updated the relevant commit now, hopefully it makes a bit more sense.